packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -891,22 +901,31 @@ export class Project extends ProjectBase {
         'ec2:DescribeSubnets',
         'ec2:DescribeSecurityGroups',
         'ec2:DescribeDhcpOptions',
-        'ec2:DescribeVpcs'
+        'ec2:DescribeVpcs',
       ));
-    this.addToRolePolicy(new iam.PolicyStatement()
-      .addResource(`arn:aws:ec2:${Aws.region}:${Aws.accountId}:network-interface/*`)
-      .addCondition('StringEquals', {
-        "ec2:Subnet": props.vpc
-          .selectSubnets(props.subnetSelection).subnetIds
-          .map(si => `arn:aws:ec2:${Aws.region}:${Aws.accountId}:subnet/${si}`),
-        "ec2:AuthorizedService": "codebuild.amazonaws.com"
-      })
-      .addAction('ec2:CreateNetworkInterfacePermission'));
-    return {
-      vpcId: props.vpc.vpcId,
-      subnets: props.vpc.selectSubnets(props.subnetSelection).subnetIds,
-      securityGroupIds: this._securityGroups.map(s => s.securityGroupId)
-    };
+  }
+
+  private addInlineEc2Policy(statement: iam.PolicyStatement) {
+    if (this.role) {
+      // If the Role is imported, the Policy will be left dangling,
+      // not attached to any IAM entity, which is invalid in CFN.
+      // Because of that, check for that case before creating the Policy
+      const cfnRole = this.role.node.tryFindChild('Resource') as CfnResource;
+      if (cfnRole) {
+        const policy = new iam.Policy(this, 'PolicyDocument', {
+          policyName: 'CodeBuildEC2Policy',
+          statements: [statement]
+        });
+        this.role.attachInlinePolicy(policy);
+
+        // add an explicit dependency between the EC2 Policy and this Project -
+        // otherwise, creating the Project fails,
+        // as it requires these permissions to be already attached to the Project's Role
+        const cfnPolicy = policy.node.findChild('Resource') as CfnResource;
+        const cfnProject = this.node.findChild('Resource') as CfnResource;
+        cfnProject.addDependsOn(cfnPolicy);
```

Comment:
`this.addDependsOn(policy)` should also do this (and more). Any reason you can't use it?

Reply:
The `Project` class doesn't have an `addDependsOn()` method.

Reply:
Okay, I guess `cfnProject.addDependsOn(policy)` then.

But `this.node.findChild('Resource')` should never be necessary. We made that child, so we could also just keep a reference to it. So:

```
this.project.addDependsOn(policy);
```
?

Reply:
Heh, that's funny. I actually added a field for the L1 initially, but then I removed it because it was used in only this single place.

I'll add it back if you think that's better.

Reply:
Done.